plugins/chat/assets/javascripts/discourse/components/chat-channel.js

```diff
@@ -113,6 +113,13 @@ export default class ChatLivePane extends Component {
       return;
     }
 
+    if (
+      this.args.channel.isDirectMessageChannel &&
+      !this.args.channel.isFollowing
+    ) {
+      this.chatChannelsManager.follow(this.args.channel);
+    }
+
```

Comment:
This is the second important change, basically opening a DM automatically follow it (client side) which prevents the green dot flashing when sending the message